geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaDataStoreParams.scala

```diff
@@ -105,6 +105,14 @@ object KafkaDataStoreParams extends NamespaceParams {
       readWrite = ReadWriteFlag.WriteOnly
     )
 
+  val OnSchemaDeleteTruncate =
+    new GeoMesaParam[java.lang.Boolean](
+      "kafka.catalog.on-schema-delete-truncate",
```

Comment:
can we rename this `kafka.topic.truncate-on-delete`? we're not actually deleting the catalog topic here, just the data topic.

similarly for the related variable names, `truncateOnDelete` is a little cleaner/clearer, in my mind